core/src/main/java/org/apache/struts2/components/Link.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.struts2.components;
+
+import com.opensymphony.xwork2.util.ValueStack;
+import org.apache.struts2.views.annotations.StrutsTag;
+import org.apache.struts2.views.annotations.StrutsTagAttribute;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * <!-- START SNIPPET: javadoc -->
+ * <p>
+ * Add nonce propagation feature to implement CSP in link tags
+ * </p>
+ *
+ * <p>
+ * The link tag allows the user to load external resources, most usually style sheets. External resources
+ * can inject malicious code and perform XSS and data injection attacks. The s:link tag includes a nonce
+ * attribute that is being randomly generated with each request and only allows links with the valid
+ * nonce value to be executed.
+ * </p>
+ * <!-- END SNIPPET: javadoc -->
+ *
+ * <p><b>Examples</b></p>
+ *
+ * <pre>
+ * <!-- START SNIPPET: example -->
+ *
+ * &lt;s:link ... /&gt;
+ *
+ * <!-- END SNIPPET: example -->
+ * </pre>
+ *
+ */
+@StrutsTag(name="link",
+        tldTagClass="org.apache.struts2.views.jsp.ui.LinkTag",
+        description="Link tag automatically adds nonces to link blocks - should be used in combination with Strut's CSP Interceptor.",
+        allowDynamicAttributes=true)
+public class Link extends UIBean{
+
+    private static final String TEMPLATE="link";
+
+    protected String href;
+    protected String hreflang;
+    protected String rel;
+    protected String media;
+    protected String referrerpolicy;
+    protected String sizes;
+    protected String crossorigin;
+    protected String type;
+    protected String as;
+    protected String importance; //docs say importance is an experimental API TODO: should we include it
+    protected String integrity; //experimental API too
```

Comment:
if these two fields are not production ready, I'd argue we should remove them

Reply:
makes sense, the [specs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/link#Attributes) say it shouldn't be used in production so I'll remove them!